### PR TITLE
🔥 Add external credentials

### LINKED ISSUE #210: Allow Multiple Proxmox Clusters
**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

We need to make credentials part of the ProxmoxCluster CR so we can enable supporting many proxmox clusters

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-proxmox version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by mcbenjemaa**
This PR adds external credentials to the Cluster.
To make it able to use multiple Proxmox data centers

Close #210

### CODE REVIEW

File: pkg/scope/cluster.go [hunk 1]
```diff
@@ -19,18 +19,27 @@ package scope
...
+	clustererrors "sigs.k8s.io/cluster-api/errors"
 	"sigs.k8s.io/cluster-api/util/conditions"
 	"sigs.k8s.io/cluster-api/util/patch"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 
 	infrav1alpha1 "github.com/ionos-cloud/cluster-api-provider-proxmox/api/v1alpha1"
 	"github.com/ionos-cloud/cluster-api-provider-proxmox/pkg/kubernetes/ipam"
-	"github.com/ionos-cloud/cluster-api-provider-proxmox/pkg/proxmox"
+	capmox "github.com/ionos-cloud/cluster-api-provider-proxmox/pkg/proxmox"
```
Comment (wikkyk) [inline pkg/scope/cluster.go:41]: is it necessary to override the name here?

File: api/v1alpha1/proxmoxcluster_types.go [hunk 2]
```diff
@@ -71,6 +72,11 @@ type ProxmoxClusterSpec struct {
 	// in the configuration and cloning of a proxmox VM. Multiple types of nodes can be specified.
 	// +optional
 	CloneSpec *ProxmoxClusterCloneSpec `json:"cloneSpec,omitempty"`
+
+	// CredentialsRef is a reference to a Secret that contains the credentials to use for provisioning this cluster. If not
+	// supplied then the credentials of the controller will be used.
+	// +optional
+	CredentialsRef *corev1.ObjectReference `json:"credentialsRef,omitempty"`
```
Comment (lubedacht) [inline api/v1alpha1/proxmoxcluster_types.go:79]: Maybe `SecretReference` or if you expect the secret to be in the same namespace `LocalObjectReference` would be sufficient?

Comment (mcbenjemaa) [reply api/v1alpha1/proxmoxcluster_types.go]: SecretReference is okay,

File: pkg/scope/cluster.go (hunk 1 above)
Comment (mcbenjemaa) [reply pkg/scope/cluster.go:41]: yes, cause we have 3 proxmox pkgs :D

Comment (wikkyk) [reply pkg/scope/cluster.go:41]: sure but none of them are named `proxmox`.

**Review by wikkyk — CHANGES_REQUESTED** [review]
The code is good to go in but this PR needs to update the documentation to explain how to use external credentials instead of PROXMOX_{URL,TOKEN,SECRET}.

**Comment by mcbenjemaa** [comment]
Please check the added docs.

**Review by wikkyk — APPROVED** [review]

**Pull request merged**